ethereum/core/src/test/java/org/hyperledger/besu/ethereum/core/TransactionSizeAndHashTest.java

```diff
@@ -0,0 +1,421 @@
+
+    assertThat(transaction.getHash()).isEqualTo(Hash.hash(bytes));
+
+    final Transaction transactionRead =
+        TransactionDecoder.decodeOpaqueBytes(bytes, EncodingContext.BLOCK_BODY);
+
+    assertThat(transactionRead.getSizeForAnnouncement()).isEqualTo(bytes.size());
+    assertThat(transactionRead.getSizeForBlockInclusion()).isEqualTo(bytes.size());
+
+    assertThat(transactionRead.getHash()).isEqualTo(Hash.hash(bytes));
+  }
+
+  @Test
+  public void returnsRightSizeForCodeDelegationTx() {
+    final Transaction transaction = buildCodeDelegationTx();
+    final Bytes bytes =
+        TransactionEncoder.encodeOpaqueBytes(transaction, EncodingContext.BLOCK_BODY);
+
+    assertThat(transaction.getSizeForAnnouncement()).isEqualTo(bytes.size());
+    assertThat(transaction.getSizeForBlockInclusion()).isEqualTo(bytes.size());
+
+    assertThat(transaction.getHash()).isEqualTo(Hash.hash(bytes));
+
+    final Transaction transactionRead =
+        TransactionDecoder.decodeOpaqueBytes(bytes, EncodingContext.BLOCK_BODY);
+
+    assertThat(transactionRead.getSizeForAnnouncement()).isEqualTo(bytes.size());
+    assertThat(transactionRead.getSizeForBlockInclusion()).isEqualTo(bytes.size());
+
+    assertThat(transactionRead.getHash()).isEqualTo(Hash.hash(bytes));
+  }
+
+  @Test
+  public void returnsRightSizeForAccessListTx() {
+    final Transaction transaction = buildAccessListTx();
+    final Bytes bytes =
+        TransactionEncoder.encodeOpaqueBytes(transaction, EncodingContext.BLOCK_BODY);
+
+    assertThat(transaction.getSizeForAnnouncement()).isEqualTo(bytes.size());
+    assertThat(transaction.getSizeForBlockInclusion()).isEqualTo(bytes.size());
+
+    assertThat(transaction.getHash()).isEqualTo(Hash.hash(bytes));
+
+    final Transaction transactionRead =
+        TransactionDecoder.decodeOpaqueBytes(bytes, EncodingContext.BLOCK_BODY);
+
+    assertThat(transactionRead.getSizeForAnnouncement()).isEqualTo(bytes.size());
+    assertThat(transactionRead.getSizeForBlockInclusion()).isEqualTo(bytes.size());
+
+    assertThat(transactionRead.getHash()).isEqualTo(Hash.hash(bytes));
+  }
+
+  @Test
+  public void returnsRightSizeForBlobTxWithoutBlobs() {
+    final Transaction transaction = buildBlobTransactionNoBlobs();
+    final Bytes bytes =
+        TransactionEncoder.encodeOpaqueBytes(transaction, EncodingContext.BLOCK_BODY);
+
+    // Blob transactions without blobs cannot be announced
+    assertThat(transaction.getSizeForBlockInclusion()).isEqualTo(bytes.size());
+
+    assertThat(transaction.getHash()).isEqualTo(Hash.hash(bytes));
+
+    final Transaction transactionRead =
+        TransactionDecoder.decodeOpaqueBytes(bytes, EncodingContext.BLOCK_BODY);
+
+    assertThat(transactionRead.getSizeForBlockInclusion()).isEqualTo(bytes.size());
+
+    assertThat(transactionRead.getHash()).isEqualTo(Hash.hash(bytes));
+  }
+
+  @Test
+  public void returnsRightSizeWithBlobs1a() {
```

Comment:
Think we think need a better name for this and the other numbered tests

Reply:
Done